restx-core-shell/src/main/java/restx/core/shell/DepsShellCommand.java

```diff
@@ -115,6 +121,43 @@ private void installDepsFromModuleDescriptor(RestxShell shell, File mdFile) thro
             }
         }
 
+        private void installDepsFromMavenDescriptor(RestxShell shell, File pomFile) throws Exception {
+            AppSettings appSettings = shell.getFactory().getComponent(AppSettings.class);
+
+            Path dependenciesDir = Paths.get(appSettings.targetDependency());
+
+            // Emptying target dependencies directory first
+            if(dependenciesDir.toFile().exists()) {
+                java.nio.file.Files.walkFileTree(dependenciesDir, new SimpleFileVisitor<Path>() {
+                    @Override
+                    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                        java.nio.file.Files.delete(file);
+                        return FileVisitResult.CONTINUE;
+                    }
+                    @Override
+                    public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+                        java.nio.file.Files.delete(dir);
+                        return FileVisitResult.CONTINUE;
+                    }
+                });
+            }
+
+            dependenciesDir.toFile().mkdirs();
+
+            // Then copying dependencies through copy-dependencies plugin
+            ProcessBuilder mavenCmd = new ProcessBuilder(
+                    "mvn", "org.apache.maven.plugins:maven-dependency-plugin:3.0.1:copy-dependencies",
+                    "-DoutputDirectory=" + dependenciesDir.toAbsolutePath(), "-DincludeScope=runtime"
+            );
+
+            shell.println("Executing `"+mavenCmd+"` ...");
+            mavenCmd.redirectErrorStream(true)
+             .redirectOutput(ProcessBuilder.Redirect.INHERIT)
+             .directory(shell.currentLocation().toFile().getAbsoluteFile())
+             .start()
+             .waitFor();
```

Comment:
How does this behave if we don't have maven installed? I know it will fail, but is the error message clear enough? And when it fails we could try to use `Ivy` to resolve the dependencies (as `Ivy` supports maven files, and is already a dependency of restx shell). With `Ivy` the code would be very similar to the code used with `md.restx.json` files, eg sg like:
```
                shell.println("resolving dependencies...");
                ResolveReport resolveReport = ivy.resolve(tempFile);

                shell.println("synchronizing dependencies in " + shell.currentLocation().resolve("target/dependency") + " ...");
                ivy.retrieve(resolveReport.getModuleDescriptor().getModuleRevisionId(),
                        new RetrieveOptions()
                                .setDestArtifactPattern(
                                        shell.currentLocation().toAbsolutePath() + "/target/dependency/[artifact]-[revision](-[classifier]).[ext]")
                        .setSync(true)
                );

                File md5File = md5File(shell);
                Files.write(Files.hash(mdFile, Hashing.md5()).toString(), md5File, Charsets.UTF_8);

                shell.println("DONE");
```

WDYT?

Reply:
Here is the console output when `mvn` CLI is not found :
```
> deps install
installing deps using maven descriptor...
Executing `java.lang.ProcessBuilder@18f8cd79` ...
Exception in thread "main" java.io.IOException: Cannot run program "mvn" (in directory "/private/tmp/bla/srv"): error=2, No such file or directory
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
	at restx.core.shell.DepsShellCommand$InstallDepsCommandRunner.installDepsFromMavenDescriptor(DepsShellCommand.java:165)
	at restx.core.shell.DepsShellCommand$InstallDepsCommandRunner.run(DepsShellCommand.java:90)
	at restx.shell.RestxShell.doExec(RestxShell.java:369)
	at restx.shell.RestxShell.exec(RestxShell.java:132)
	at restx.shell.RestxShell.main(RestxShell.java:415)
```
I'd say it is reasonably clear to my POV.

> When it fails we could try to use Ivy to resolve the dependencies (as Ivy supports maven files, and is already a dependency of restx shell).

I'm not a big fan of this. 

To my POV, the probability that :
- Current project has a `pom.xml` AND doesn't have a `md.restx.json`
AND
- Current user has ivy installed AND doesn't have maven installed

... is very low.

Even if your proposition is making the shell smarter, it would complexify the deps installation from maven with code which is, from my opinion, very unlikely to be executed.

Moreover, what I'm most afraid of is detecting that `mvn` CLI is missing. As you can see above, an `IOException` is thrown, and we should have to look at this exception message to determine if this is a `mvn CLI is missing`-class-of-exceptions.

Really, not a big fan of this.

Reply:
Ivy doesn't have to be "installed", it's part of the dependencies of restx shell (it's used to provision shell plugins), so it should always be available.

The use case I see is when you use these commands on a server to run the app: you may prefer to use only `pom.xml` by habits, and still in most cases Ivy will be able to tackle the task of dependencies provisionning on a server where you want to run your app and don't want to install maven.

I agree the detection is not easy, so maybe it could be an option on the command, and when catching an exception when launching maven we could simply log an information to suggest using the option.

WDYT?

Reply:
OK, I added an exception message parsing to detect if `mvn` command is missing and, in that case, automatically falling back on Ivy resolution :
```
            try {
                mavenCmd.redirectErrorStream(true)
                        .redirectOutput(ProcessBuilder.Redirect.INHERIT)
                        .directory(shell.currentLocation().toFile().getAbsoluteFile())
                        .start()
                        .waitFor();
            } catch(IOException e) {
                if(e.getMessage().startsWith(String.format("Cannot run program \"%s\"", MVN_COMMAND))) {
                    shell.println("Looks like mvn is not installed here ... trying to fallback installing pom.xml file with Ivy...");
                    installDepsFromIvyDescriptor(shell, pomFile);
                } else {
                    Throwables.propagate(e);
                }
            }
```